pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -518,13 +529,16 @@ func (vs *VSphere) AttachDisk(vmDiskPath string, nodeName string) (diskID string
 		return "", "", err
 	}
 
-	// find SCSI controller to attach the disk
-	var newSCSICreated bool = false
+	var diskControllerType = vs.cfg.Disk.SCSIControllerType
+	// find SCSI controller of particular type from VM devices
+	var diskController = getSCSIController(vmDevices, diskControllerType)
```

Comment:
FindDiskController can only find based on the controller( either scsi or ide). there is no govmomi api call to get scsi based on its driver type. so I have to add getSCSIController() to filter VM devices based on the particular scsi driver type.